Utilities/WDLs/MergeSingleSampleMinimacVcfs.wdl

```diff
@@ -0,0 +1,516 @@
+        memory: mem_gb + " GiB"
+        cpu: cpu
+        preemptible: preemptible
+		}
+
+    output {
+    	File merged_vcf = "merged.vcf.gz"
+    	File merged_vcf_index = "merged.vcf.gz.tbi"
+    }
+
+}
+
+task reannotate_from_dosages {
+    input {
+        File vcf
+        Int n_samples
+        Int disk_size_gb = ceil(3.3 * size(vcf, "GiB")) + 50
+        Int mem_gb = 4
+        Int cpu = 1
+        Int chunksize = 1000000
+        Int preemptible = 3
+    }
+
+    String output_base = basename(vcf, ".vcf.gz")
+
+    command <<<
+        set -euo pipefail
+
+        bcftools query -f '%CHROM,%POS,%REF,%ALT[,%DS]' ~{vcf} | gzip -c > dosage_tbl.csv.gz
+
+        echo "dosages extracted"
+
+        python3 << EOF
+        import pandas as pd
+        import numpy as np
+
+
+        out_annotation_dfs = []
+        for chunk in pd.read_csv("dosage_tbl.csv.gz", names=["CHROM","POS","REF","ALT"]+[f'sample_{i}' for i in range(~{n_samples})],dtype={f'sample_{i}':'float' for i in range(~{n_samples})},
+                        na_values=".", chunksize = ~{chunksize}):
+            chunk.dropna(inplace=True)
+            dosages = chunk[[f'sample_{i}' for i in range(~{n_samples})]].to_numpy()
+            chunk_annotations = chunk[["CHROM","POS","REF","ALT"]]
+            chunk_annotations["AF"] = dosages.mean(axis=1)/2
+            chunk_annotations["R2"] = np.where((chunk_annotations["AF"]==0) | (chunk_annotations["AF"]==1),1,dosages.var(axis=1)/(2*chunk_annotations["AF"]*(1-chunk_annotations["AF"])))
+            out_annotation_dfs.append(chunk_annotations)
+        annotations_df = pd.concat(out_annotation_dfs)
+        annotations_df.to_csv("annotations.tsv", sep="\t", index=False, header=False)
+        EOF
+        
+        echo "annotations recomputed"
+
+        bgzip annotations.tsv
+        tabix -s1 -b2 -e2 annotations.tsv.gz
+
+        echo "annotating vcf with new annotations"
+
+        bcftools annotate --no-version -a annotations.tsv.gz -c CHROM,POS,REF,ALT,AF,R2 -x INFO/MAF -Oz -o ~{output_base}.reannotated.vcf.gz ~{vcf}
+        
+        
+
+    >>>
+
+
+    runtime {
+        docker: "us.gcr.io/broad-dsde-methods/samtools-suite:v1.1"
+        disks: "local-disk " + disk_size_gb + " HDD"
+        memory: mem_gb + " GiB"
+        cpu: cpu
+        preemptible: preemptible
+    }
+
+    output {
+        File annotated_vcf = "~{output_base}.reannotated.vcf.gz"
+    }
+    
+}
+
+task CountSamples { # really?
+    input {
+        File imputed_vcf
+    }
+
+
+    parameter_meta {
+        imputed_vcf:
+            {
+                localization_optional: true
+            }
+    }
+
+    command <<<
+        set -xeuo pipefail
+        export GCS_OAUTH_TOKEN=$(/root/google-cloud-sdk/bin/gcloud auth application-default print-access-token)
+        bcftools query -l ~{imputed_vcf} | wc -l > "num_samples.txt"
+    >>>
+
+    runtime {
+        docker: "us.gcr.io/broad-dsde-methods/bcftools:v1.3"
```

Comment:
I know I wrote this workflow originally but I don't know how I feel about no runtime resources being defined. Maybe it's fine